test/runtests.jl

```diff
@@ -21,6 +21,53 @@ function draw2(μ)
     return (x,y)
 end
 
+measures = [
+    # Chain(x -> Normal(μ=x), Normal(μ=0.0))
+    For(3) do j Normal(σ=j) end
+    For(2,3) do i,j Normal(i,j) end
+    # Likelihood
+    # Pointwise
+    Normal() ^ 3
+    Normal() ^ (2,3)
+    # SpikeMixture(Normal(), 2)
+    # Dirac(0.0) + Normal()
+    # transforms
+    3 * Normal()
+    Bernoulli(0.2)
+    Beta(2,3)
+    Binomial(10,0.3)
+    Cauchy()
+    Dirichlet(ones(3))
+    Exponential()
+    Gumbel()
+    # InverseGamma(2)
+    Laplace()
+    LKJCholesky(3,2.0)
+    Multinomial(n=10,p=[0.2,0.3,0.5])
+    # MvNormal
+    NegativeBinomial(5,0.2)
+    Normal(2,3)
+    Poisson(3.1)
+    StudentT(ν=2.1)    
+    Uniform()
+    # CountingMeasure(Float64)
+    Dirac(π)
+    Lebesgue(ℝ)
+    # TrivialMeasure()
+]
+
+@testset "testvalue" begin
+    for μ in measures
+        @test logdensity(μ, testvalue(μ)) isa Float64
+    end
+    
+    @testset "testvalue(::Chain)" begin
+        mc =  Chain(x -> Normal(μ=x), Normal(μ=0.0))
+        r = testvalue(mc)
+        @test logdensity(mc, Iterators.take(r, 10)) isa Float64
```

Comment:
This is a little unintuitive for me. So `testvalue`, which is I think a point on the sample space of `Chain`, cannot be passed to the `logdensity` function without some processing. Would it make more sense for `Chain` to be a family of measures that must be conditioned on a fixed number of steps (here, 10), in order to uniquely define a measure?

Reply:
I agree it's unintuitive, but I think it's more a difficulty of the math than a problem with the implementation. It's a little like working with irrational numbers. Everything is well-defined, but you can never look at or do computations with the full set of decimal digits. Or even better a Gaussian process - the GP itself is a continuous function, but we can only ever evaluate it on a finite set of values.